_posts/2025-01-29-cockpit-332.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Cockpit 332
+author: jelle
+date: '2025-01-29'
+tags: cockpit, machines, podman, files
+slug: cockpit-332
+category: release
+summary: 'package-less session support for more operation systems, show system boot times in metrics, automatically start podman.socket and improve quadlet support in cockpit-podman, show user group in files footer'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 332, cockpit-machines 327, cockpit-podman 100, and cockpit-files 15:
+
+## cockpit/ws container: Include cockpit-files
+
+The [cockpit/ws container](https://quay.io/repository/cockpit/ws) now includes [cockpit-files](https://github.com/cockpit-project/cockpit-files/). When you log into a remote machine that does not have any cockpit packages installed (package-less mode), the "Files" page is now available.
+
+## Package-less session for all supported operating systems
+
+[Cockpit 327](https://cockpit-project.org/blog/cockpit-327.html) introduced support for connecting to remote machines without any installed Cockpit packages. That only worked if the local and remote machine had the same operating system version. This version extends this "package-less session" feature to all operating systems which Cockpit [continuously tests](https://cockpit-project.org/running.html#installation--setup). At the time of writing this includes:
+
+  * Arch Linux
+  * CentOS Stream 9, 10
+  * Debian 12, testing/unstable
+  * Fedora Linux 40, 41, 42 (rawhide)
+  * Red Hat Enterprise Linux 8, 9, 10
+  * Ubuntu 22.04 LTS, 24.04 LTS, 24.10
+
+This list will change over time, along with new operating system releases and their end-of-lives.
+
+## Show system boot in metrics
+
+![screenshot of ## show system boot in metrics]({{ site.baseurl }}/images/332-show-system-boot-in-metrics.png)
+
+## Machines: "Launch viewer" now uses correct address
+
+The "Launch viewer" button in the "Console" card of a virtual machine would tell the remote viewer application to connect to the listening address of the VNC server. This is wrong when the server listens on 127.0.0.1 (libvirt default) or 0.0.0.0 (allow connections on all interfaces). Now Cockpit will tell the remote viewer application to connect to the same domain name or IP address that Cockpit itself is using to reach the virtual machine host.
```

Comment:
```suggestion
The "Launch viewer" button in the "Console" card of a virtual machine would tell the remote viewer application to connect to the listening address of the VNC server. This is problematic when the server listens on 127.0.0.1 (libvirt default) or 0.0.0.0 (allow connections on all interfaces). To address this, Cockpit will now tell the remote viewer application to connect to the same domain name or IP address that Cockpit itself uses to reach the virtual machine host.
```

Reply:
I'm going to simplify this, it is too technical:


> can we just change the release note to say that we connect directly to the domain name or IP of the virtual machine host which resolves issues when someone has configured VNC/SPICE to listen to on interfaces. Note that when listening to 127.0.0.1 cockpit-machines cannot display the vnc/spice console
the current release note is rather technical and claims [localhost](http://localhost/) works